backend/src/apiserver/server/visualization_server.go

```diff
@@ -61,7 +68,11 @@ func (s *VisualizationServer) validateCreateVisualizationRequest(request *go_cli
 // service to generate HTML visualizations from a request.
 // It returns the generated HTML as a string and any error that is encountered.
 func (s *VisualizationServer) generateVisualizationFromRequest(request *go_client.CreateVisualizationRequest) ([]byte, error) {
-	if !isVisualizationServiceAlive(s.serviceURL) {
+	serviceURL, err := s.getVisualizationServiceURL(request)
```

Comment:
Before this step, I think you should check whether the user is authorized to visit that namespace

Reply:
: )